src/pycountry/__init__.py

```diff
@@ -126,101 +124,125 @@ class HistoricCountries(ExistingCountries):
     """Provides access to an ISO 3166-3 database
     (Countries that have been removed from the standard)."""
 
-    data_class = pycountry.db.Country
+    factory = pycountry.db.Country
     root_key = "3166-3"
 
 
-class Scripts(pycountry.db.Database):
+class Script(pycountry.db.Data):
+    pass
+
+
+class Scripts(pycountry.db.Database[Script]):
     """Provides access to an ISO 15924 database (Scripts)."""
 
-    data_class = "Script"
+    factory = Script
     root_key = "15924"
 
 
-class Currencies(pycountry.db.Database):
+class Currency(pycountry.db.Data):
+    pass
+
+
+class Currencies(pycountry.db.Database[Currency]):
     """Provides access to an ISO 4217 database (Currencies)."""
 
-    data_class = "Currency"
+    factory = Currency
     root_key = "4217"
 
 
-class Languages(pycountry.db.Database):
+class Language(pycountry.db.Data):
+    pass
+
+
+class Languages(pycountry.db.Database[Language]):
     """Provides access to an ISO 639-1/2T/3 database (Languages)."""
 
     no_index = ["status", "scope", "type", "inverted_name", "common_name"]
 
-    data_class = "Language"
+    factory = Language
     root_key = "639-3"
 
 
-class LanguageFamilies(pycountry.db.Database):
+class LanguageFamily(pycountry.db.Data):
+    pass
+
+
+class LanguageFamilies(pycountry.db.Database[LanguageFamily]):
     """Provides access to an ISO 639-5 database
     (Language Families and Groups)."""
 
-    data_class = "LanguageFamily"
+    factory = LanguageFamily
     root_key = "639-5"
 
 
 class SubdivisionHierarchy(pycountry.db.Data):
-    def __init__(self, **kw):
-        if "parent" in kw:
-            kw["parent_code"] = kw["parent"]
-        else:
-            kw["parent_code"] = None
-        super().__init__(**kw)
-        self.country_code = self.code.split("-")[0]
-        if self.parent_code is not None:
-            # Split the parent_code to check if the country_code is already present
-            parts = self.parent_code.split("-")
-            if parts[0] != self.country_code:
-                self.parent_code = f"{self.country_code}-{self.parent_code}"
-
     @property
-    def country(self):
-        return countries.get(alpha_2=self.country_code)
+    def country(self) -> pycountry.db.Country:
+        return cast(
+            pycountry.db.Country, countries.get(alpha_2=self.country_code)
+        )
```

Comment:
This can also be typed:

```suggestion
        result = countries.get(alpha_2=self.country_code)
        assert result
        return result
```